resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -784,6 +794,9 @@ private[spark] class Client(
     val env = new HashMap[String, String]()
     populateClasspath(args, hadoopConf, sparkConf, env, sparkConf.get(DRIVER_CLASS_PATH))
     env("SPARK_YARN_STAGING_DIR") = stagingDirPath.toString
+    if (isClientUnmanagedAMEnabled) {
+      System.setProperty("SPARK_YARN_STAGING_DIR", stagingDirPath.toString)
```

Comment:
Can this be propagated some other way? Using system properties is kinda hacky, and makes it dangerous to run another Spark app later in the same JVM.

Reply:
Changed it to get from the spark conf and the application id.